### PR TITLE
add aes256gcm / sha384 support

### PR DESCRIPTION
**Pull request by kazuho**
The most-preferred cipher-suite becomes AES256GCM as well.

### CODE REVIEW

File: t/picotls.c [hunk 1]
```diff
@@ -735,9 +753,11 @@ static void test_stateless_hrr_aad_change(void)
 void test_picotls(void)
 {
     subtest("sha256", test_sha256);
+    subtest("sha384", test_sha384);
```
Comment (huitema) [inline t/picotls.c]: I was wondering why the port to Windows still compiled, without me having added sha512.c to the VS project. But now I get it: there is no test for sha512, thus no binary dependency!

Comment (huitema) [reply t/picotls.c]: But actually, sha512.c is already listed in the VS project. It is not a new file. SO we are good. Although we may want to add a test...

Comment (huitema) [reply t/picotls.c]: And of course no ciphersuite uses it, so there is no easy way to write a test...

Comment (kazuho) [reply t/picotls.c]: aes256gcmsha384 uses it :-)

**Pull request merged**